extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetailsHeader.ts

```diff
@@ -110,17 +118,44 @@ export class AssessmentDetailsHeader {
 			valueContainer.value = assessmentHeaderValues[index++].value);
 	}
 
-	// function that returns a localised string for target platform based on enum MigrationTargetType.
-	private _getTargetPlatformLabel(targetType: MigrationTargetType): string {
+	// function that create target selection dropdown
+	private createTargetTypeContainer(): azdata.FlexContainer {
+		const targetTypeContainer = this._view.modelBuilder.flexContainer().component();
+
+		const selectLabel = this._view.modelBuilder.text().withProps({
+			value: constants.SELECT_TARGET_LABEL + ":",
+			CSSStyles: {
+				...styles.LABEL_CSS
+			},
+		}).component();
+
+		this._targetSelectionDropdown = this._view.modelBuilder.dropDown().withProps({
+			ariaLabel: constants.AZURE_SQL_TARGET,
+			value: constants.SUMMARY_SQLDB_TYPE,
+			values: [constants.SUMMARY_SQLDB_TYPE, constants.SUMMARY_VM_TYPE, constants.SUMMARY_MI_TYPE],
+			width: 250,
+			editable: false,
+			CSSStyles: {
+				'margin-top': '-0.2em',
+				'margin-left': '10px'
+			},
+		}).component();
+
+		targetTypeContainer.addItem(selectLabel, { flex: 'none' });
+		targetTypeContainer.addItem(this._targetSelectionDropdown, { flex: 'none' });
+
+		return targetTypeContainer;
+	}
+
+	// function to get target type value based on model value.
+	private getTargetTypeBasedOnModel(targetType: MigrationTargetType): string {
 		switch (targetType) {
 			case MigrationTargetType.SQLDB:
-				return constants.SUMMARY_SQLDB_TYPE
+				return constants.SUMMARY_SQLDB_TYPE;
 			case MigrationTargetType.SQLVM:
-				return constants.SUMMARY_VM_TYPE
+				return constants.SUMMARY_VM_TYPE;
 			case MigrationTargetType.SQLMI:
-				return constants.SUMMARY_MI_TYPE
-			default:
-				return "--"
+				return constants.SUMMARY_MI_TYPE;
```

Comment:
Why in default case we are sending MI Summary?

Reply:
there is no default value, only three cases DB, MI and VM. This is an enum and there are only three values in enum. I believe we don't need any default case here. Correct me if needed.

Reply:
In that case, we should keep returning as earlier. Otherwise in some error scenario it will return MI and we won't be able to catch it.  So, either throw exception in default case or return unsupported type.

Reply:
added a default case to throw exception of unsupported type.